src/lib/CacheInstance.ts

```diff
@@ -27,6 +26,17 @@ export abstract class CacheInstance extends EventEmitter {
    */
   public abstract getValue(key: string): Promise<CachableValue>;
 
+  /**
+   * Find all keys matching the given pattern
```

Comment:
Please document the O(n)-ness!
```suggestion
   * Find all keys matching the given pattern.
   *
   * PERFORMANCE WARNING: this is O(keys) / will TRAVERSE ALL KEYS!
   * Both our Local & Redis implems can traverse a lot of keys quickly,
   * but be extremely careful calling this! This should be used for
   * occasional operations, not for high-frequency calls!
```